packages/aws-cdk/lib/cli/telemetry/endpoint-sink.ts

```diff
@@ -0,0 +1,141 @@
+import type { IncomingMessage } from 'http';
+import type { Agent } from 'https';
+import { request } from 'https';
+import type { UrlWithStringQuery } from 'url';
+import { ToolkitError } from '@aws-cdk/toolkit-lib';
+import { IoHelper } from '../../api-private';
+import type { IIoHost } from '../io-host';
+import type { TelemetrySchema } from './schema';
+import type { ITelemetrySink } from './sink-interface';
+
+const REQUEST_ATTEMPT_TIMEOUT_MS = 500;
+
+/**
+ * Properties for the Endpoint Telemetry Client
+ */
+export interface EndpointTelemetrySinkProps {
+  /**
+   * The external endpoint to hit
+   */
+  readonly endpoint: UrlWithStringQuery;
+
+  /**
+   * Where messages are going to be sent
+   */
+  readonly ioHost: IIoHost;
+
+  /**
+   * The agent responsible for making the network requests.
+   *
+   * Use this to set up a proxy connection.
+   *
+   * @default - Uses the shared global node agent
+   */
+  readonly agent?: Agent;
+}
+
+/**
+ * The telemetry client that hits an external endpoint.
+ */
+export class EndpointTelemetrySink implements ITelemetrySink {
+  private events: TelemetrySchema[] = [];
+  private endpoint: UrlWithStringQuery;
+  private ioHelper: IoHelper;
+  private agent?: Agent;
+
+  public constructor(props: EndpointTelemetrySinkProps) {
+    this.endpoint = props.endpoint;
+    this.ioHelper = IoHelper.fromActionAwareIoHost(props.ioHost);
+    this.agent = props.agent;
+
+    // Batch events every 30 seconds
+    setInterval(() => this.flush(), 30000).unref();
+  }
+
+  /**
+   * Add an event to the collection.
+   */
+  public async emit(event: TelemetrySchema): Promise<void> {
+    try {
+      this.events.push(event);
+    } catch (e: any) {
+      // Never throw errors, just log them via ioHost
+      await this.ioHelper.defaults.trace(`Failed to add telemetry event: ${e.message}`);
+    }
+  }
+
+  public async flush(): Promise<void> {
+    try {
+      if (this.events.length === 0) {
+        return;
+      }
+
+      const res = await this.https(this.endpoint, this.events);
```

Comment:
Should be:

```suggestion
      const res = await this.https(this.endpoint, { events: this.events });
```

Because this is that server (will) expect.